rotkehlchen/exchanges/woo.py

```diff
@@ -0,0 +1,431 @@
+KNOWN_NON_ASSET_KEYS_FOR_MOVEMENTS = {
+    'datetime',
+    'id',
+    'type',
+    'fee',
+}
+
+
+class TradePairData(NamedTuple):
+    pair: str
+    base_asset_symbol: str
+    quote_asset_symbol: str
+    base_asset: AssetWithOracles
+    quote_asset: AssetWithOracles
+
+
+class Woo(ExchangeInterface):
+    """Woo exchange api docs:
+    https://docs.woo.org/#general-information
+
+    """
+    def __init__(
+            self,
+            name: str,
+            api_key: ApiKey,
+            secret: ApiSecret,
+            database: 'DBHandler',
+            msg_aggregator: MessagesAggregator,
+    ):
+        super().__init__(
+            name=name,
+            location=Location.WOO,
+            api_key=api_key,
+            secret=secret,
+            database=database,
+        )
+        self.base_uri = 'https://api.woo.org'
+        self.msg_aggregator = msg_aggregator
+        # NB: x-api-signature & x-api-timestamp change per request
+        # x-api-key is constant
+        self.session.headers.update({
+            'x-api-key': self.api_key,
+        })
+
+    def first_connection(self) -> None:
+        self.first_connection_made = True
+
+    def edit_exchange_credentials(self, credentials: ExchangeAuthCredentials) -> bool:
+        changed = super().edit_exchange_credentials(credentials)
+        if credentials.api_key is not None:
+            self.session.headers.update({'x-api-key': credentials.api_key})
+        return changed
+
+    @protect_with_lock()
+    @cache_response_timewise()
+    def query_balances(self) -> ExchangeQueryBalances:
+        """Return the account balances on Woo """
+        response = self._api_query('v3/balances')
+
+        if response.status_code != HTTPStatus.OK:
+            result, msg = self._process_unsuccessful_response(
+                response=response,
+                case='balances',
+            )
+            return result, msg
+        try:
+            response_dict = jsonloads_dict(response.text)
+        except JSONDecodeError as e:
+            msg = f'Woo returned invalid JSON response: {response.text}.'
+            log.error(msg)
+            raise RemoteError(msg) from e
+        account_balances = self._deserialize_accounts_balances(response_dict=response_dict)
+        return account_balances, ''
+
+    def _deserialize_accounts_balances(
+            self,
+            response_dict: dict[str, Any],
+    ) -> dict[AssetWithOracles, Balance]:
+        try:
+            balances = response_dict['data']['holding']
+        except KeyError as e:
+            msg = 'Woo balances JSON response is missing data key'
+            log.error(msg, response_dict)
+            raise RemoteError(msg) from e
+
+        assets_balance: defaultdict[AssetWithOracles, Balance] = defaultdict(Balance)
+        for entry in balances:
+            symbol = entry.get('token')
+            try:
+                amount = deserialize_asset_amount(entry.get('holding') + entry.get('staked'))
```

Comment:
if you don't set a default value `.get` will return `None` if the key doesn't exists. I believe you want `0` here